### PR TITLE
feat: add storageretryAttempts to `initialize()`

### PR DESCRIPTION
**Pull request by dshukertjr**
## What kind of change does this PR introduce?

- Adds `storageRetryAttempts`. 
- Adds some docs.

### CODE REVIEW

File: lib/src/supabase.dart [hunk 1]
```diff
@@ -39,15 +39,34 @@ class Supabase {
...
+    String? schema,
+    Map<String, String>? headers,
     String? authCallbackUrlHostname,
-    bool? debug,
     LocalStorage? localStorage,
     Client? httpClient,
-    Map<String, String>? headers,
-    String? schema,
+    int storageRetryAttempts = 0,
+    bool debug = kDebugMode,
```
Comment (Vinzent03) [inline lib/src/supabase.dart:69]: This could be a breaking change

Comment (bdlukaa) [reply lib/src/supabase.dart:69]: Why? `headers` and `schema` were moved to the top of the parameters list

**Review by bdlukaa — APPROVED** [review]

File: lib/src/supabase.dart (hunk 1 above)
Comment (Vinzent03) [reply lib/src/supabase.dart:69]: I meant the `debug` parameter. Previously you could assign `null`, but now it has to be non null bool.

Comment (dshukertjr) [reply lib/src/supabase.dart]: @Vinzent03 Reverted!

**Review by Vinzent03 — APPROVED** [review]

**Pull request merged**